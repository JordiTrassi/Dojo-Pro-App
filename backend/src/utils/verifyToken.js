const jwt = require('jsonwebtoken');

function verifyToken(err, user, res) {
  if (err) {
    return res.sendStatus(403);
  }

  // eslint-disable-next-line no-underscore-dangle
  const data = { _id: user._id, email: user.email };

  const token = jwt.sign(
    { user: data },
    process.env.JWT_SECRET,
    { expiresIn: process.env.TOKEN_EXPIRES_IN },
  );

  return res.json({
    token,
  });
}

module.exports = {
  verifyToken,
};
