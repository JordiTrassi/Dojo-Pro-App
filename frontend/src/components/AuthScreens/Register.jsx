/* eslint-disable react/prop-types */
import React, { useState } from 'react';
import {
  SafeAreaView,
  View,
  ImageBackground,
  Image,
  Text,
  TextInput,
  TouchableOpacity,
} from 'react-native';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import Toast from 'react-native-simple-toast';
import { size } from 'lodash';
import validateEmail from '../../utils/validateEmail';
import confirm from '../../redux/actions/actionRegister';
import styles from './Register_Style';
import { imageBckGrd, titleImage } from '../../assets/ImagesAndIcons';

export default function Register({ navigation }) {
  const onPressMain = () => { navigation.navigate('MainScreen'); };

  const dispatch = useDispatch();
  const [name, setName] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [confirmPassword, setConfirmPassword] = useState('');

  const validateData = () => {
    let isValid = true;

    if (name.length === 0 || name.length > 10) {
      Toast.show('Minim Name characters 2 and maxim 10');
      isValid = false;
    }
    if (!validateEmail(email)) {
      Toast.show('The Email is not valid.');
      isValid = false;
    }
    if (size(password) < 6) {
      Toast.show('Password is short, minimum 6 characters');
      isValid = false;
    }
    if (size(confirmPassword) < 6) {
      Toast.show('Confirm password is short, minimum 6 characters');
      isValid = false;
    }
    if (password !== confirmPassword) {
      Toast.show('Password and Confirm Password are diferent');
      isValid = false;
    }

    return isValid;
  };

  function handleRegister() {
    if (!validateData()) {
      return;
    }
    dispatch(confirm(name, email, password));
    navigation.navigate('LoginScreen');
  }

  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.image_Stack}>
        <ImageBackground
          source={imageBckGrd}
          resizeMode="contain"
          style={styles.image1}
          imageStyle={styles.image1_imageStyle}
        >
          <TouchableOpacity
            testID="mainScreenButton"
            onPress={onPressMain}
          >
            <Image source={titleImage} style={styles.titleimage} />
          </TouchableOpacity>
          <Text style={styles.user_name_title}>User Name</Text>
          <View style={styles.rect_name}>
            <TextInput
              style={styles.input_user_name}
              onChangeText={(userName) => setName(userName)}
              value={name}
              placeholder="Name"
              autoCapitalize="none"
              keyboardType="email-address"
            />
          </View>

          <Text style={styles.email_title}>E-mail</Text>
          <View style={styles.rect_email}>
            <TextInput
              style={styles.input_email}
              onChangeText={(userEmail) => setEmail(userEmail)}
              value={email}
              placeholder="user@email.com"
              autoCapitalize="none"
              keyboardType="email-address"
            />
          </View>

          <Text style={styles.password_title}>Password</Text>
          <View style={styles.rect_password}>
            <TextInput
              style={styles.input_password}
              onChangeText={(pass) => setPassword(pass)}
              value={password}
              secureTextEntry
              placeholder="*******************"
              autoCapitalize="none"
            />
          </View>
          <Text style={styles.confirm_password_title}>Confirm Password</Text>
          <View style={styles.rect_password_two}>
            <TextInput
              style={styles.input_password}
              onChangeText={(pass2) => setConfirmPassword(pass2)}
              value={confirmPassword}
              secureTextEntry
              placeholder="*******************"
              autoCapitalize="none"
            />
          </View>

        </ImageBackground>
        <TouchableOpacity
          testID="confirmBtn"
          onPress={() => handleRegister()}
          style={styles.button_confirm}
        >
          <Text style={styles.confirmBtn}>Confirm</Text>
        </TouchableOpacity>
      </View>
    </SafeAreaView>
  );
}

Register.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
  }).isRequired,
};
