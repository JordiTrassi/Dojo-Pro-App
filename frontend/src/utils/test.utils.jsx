/* eslint-disable react/prop-types */
import React from 'react';
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import configureStore from '../redux/store';

const customRender = (ui, initialState, options) => {
  const AllTheProviders = ({ children }) => (
    <Provider store={configureStore(initialState)}>
      { children }
    </Provider>
  );

  return render(ui, {
    wrapper: AllTheProviders,
    ...options,
  });
};

// re-export everything
export * from '@testing-library/react-native';

// override render method
export { customRender as render };
