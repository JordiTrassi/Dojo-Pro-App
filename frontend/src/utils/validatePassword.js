export default function validatePassword(password) {
  const regex = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)[a-zA-Z\d]{8,}$/;
  return regex.test(password);
}
