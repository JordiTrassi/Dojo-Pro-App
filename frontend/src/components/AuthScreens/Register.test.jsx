/* eslint-disable no-undef */
import React from 'react';
import { render, fireEvent } from '../../utils/test.utils';
import Register from './Register';

jest.mock('react-native-simple-toast', () => ({
  SHORT: jest.fn(),
}));

const navigation = {
  navigate: jest.fn(),
  push: jest.fn(),
};
jest.mock('react-native-simple-toast');
const handleLogIn = jest.fn();
describe('Given a Character Component', () => {
  let screen;
  beforeEach(() => {
    screen = render(<Register navigation={navigation} />);
  });
  describe('When it is rendered', () => {
    test('Then screen should match the snapshot', () => {
      expect(screen).toMatchSnapshot();
    });
  });
  describe('And email input text is changed', () => {
    test('Then the text typed should be rendered', () => {
      const emailInput = screen.getByTestId('userInputEmail');
      fireEvent(emailInput, 'email');
      fireEvent.changeText(emailInput, 'email@email.com');
      expect(screen.queryByDisplayValue(/email@email.com/i)).not.toBe(null);
    });
  });
  describe('And you click on confirmButton', () => {
    test('Then navigation.push should have been called', () => {
      const confirmButton = screen.getByTestId('confirmBtn');
      fireEvent.press(confirmButton);
      expect(handleLogIn).toHaveBeenCalled();
    });
  });
  describe('And you click on mainScreenButton', () => {
    test('Then navigation.push should have been called', () => {
      const mainScreenButton = screen.getByTestId('mainScreenButton');
      fireEvent.press(mainScreenButton);
      expect(screen).toHaveBeenCalled();
    });
  });
});
