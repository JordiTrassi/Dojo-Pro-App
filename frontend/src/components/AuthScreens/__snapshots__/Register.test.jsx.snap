// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Character Component When it is rendered Then screen should match the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "height": 761,
        "marginLeft": -242,
        "width": 953,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "height": 819,
          "left": 0,
          "position": "absolute",
          "top": -100,
          "width": 933,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "uri": "https://www.robinju.com/wp-content/uploads/2012/10/Fushimi-Inari-011.jpg",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": 819,
              "width": 933,
            },
            Object {
              "opacity": 0.55,
            },
          ]
        }
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="mainScreenButton"
      >
        <Image
          source={
            Object {
              "uri": "https://fontmeme.com/permalink/210902/8337e2e2aeed704e1256a194f328c52c.png",
            }
          }
          style={
            Object {
              "height": 50,
              "left": 265,
              "top": 150,
              "width": 350,
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "color": "#121212",
            "fontFamily": "arial",
            "fontSize": 25,
            "left": 276,
            "position": "absolute",
            "top": 250,
          }
        }
      >
        User Name
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#E6E6E6",
            "borderColor": "#000000",
            "borderRadius": 7,
            "borderWidth": 2,
            "height": 45,
            "left": 257,
            "position": "absolute",
            "top": 292,
            "width": 300,
          }
        }
      >
        <TextInput
          autoCapitalize="none"
          keyboardType="email-address"
          onChangeText={[Function]}
          placeholder="Name"
          style={
            Object {
              "color": "#121212",
              "fontFamily": "arial",
              "fontSize": 25,
              "marginLeft": 19,
              "marginTop": 7,
            }
          }
          value=""
        />
      </View>
      <Text
        style={
          Object {
            "color": "#121212",
            "fontFamily": "arial",
            "fontSize": 25,
            "left": 276,
            "position": "absolute",
            "top": 354,
          }
        }
      >
        E-mail
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#E6E6E6",
            "borderColor": "#000000",
            "borderRadius": 7,
            "borderWidth": 2,
            "height": 45,
            "left": 257,
            "position": "absolute",
            "top": 389,
            "width": 300,
          }
        }
      >
        <TextInput
          autoCapitalize="none"
          keyboardType="email-address"
          onChangeText={[Function]}
          placeholder="user@email.com"
          style={
            Object {
              "color": "#121212",
              "fontFamily": "arial",
              "fontSize": 25,
              "marginLeft": 19,
              "marginTop": 7,
            }
          }
          value=""
        />
      </View>
      <Text
        style={
          Object {
            "color": "#121212",
            "fontFamily": "arial",
            "fontSize": 25,
            "left": 276,
            "position": "absolute",
            "top": 452,
          }
        }
      >
        Password
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#E6E6E6",
            "borderColor": "#000000",
            "borderRadius": 7,
            "borderWidth": 2,
            "height": 45,
            "left": 257,
            "position": "absolute",
            "top": 489,
            "width": 300,
          }
        }
      >
        <TextInput
          autoCapitalize="none"
          onChangeText={[Function]}
          placeholder="*******************"
          secureTextEntry={true}
          style={
            Object {
              "color": "#121212",
              "fontFamily": "arial",
              "fontSize": 25,
              "marginLeft": 19,
              "marginTop": 7,
            }
          }
          value=""
        />
      </View>
      <Text
        style={
          Object {
            "color": "#121212",
            "fontFamily": "arial",
            "fontSize": 25,
            "left": 276,
            "top": 500,
          }
        }
      >
        Confirm Password
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#E6E6E6",
            "borderColor": "#000000",
            "borderRadius": 7,
            "borderWidth": 2,
            "height": 45,
            "left": 257,
            "position": "absolute",
            "top": 586,
            "width": 300,
          }
        }
      >
        <TextInput
          autoCapitalize="none"
          onChangeText={[Function]}
          placeholder="*******************"
          secureTextEntry={true}
          style={
            Object {
              "color": "#121212",
              "fontFamily": "arial",
              "fontSize": 25,
              "marginLeft": 19,
              "marginTop": 7,
            }
          }
          value=""
        />
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#E6E6E6",
          "borderColor": "#000000",
          "borderRadius": 20,
          "borderWidth": 3,
          "elevation": 5,
          "height": 65,
          "left": 370,
          "opacity": 1,
          "position": "absolute",
          "shadowColor": "rgba(0,0,0,1)",
          "shadowOffset": Object {
            "height": 3,
            "width": 3,
          },
          "shadowOpacity": 0.35,
          "shadowRadius": 0,
          "top": 650,
          "width": 135,
        }
      }
      testID="confirmBtn"
    >
      <Text
        style={
          Object {
            "color": "#121212",
            "fontFamily": "arial",
            "fontSize": 28,
            "marginTop": 13,
            "textAlign": "center",
          }
        }
      >
        Confirm
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
